src/main/scala/no/ndla/network/User.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Part of NDLA network.
+ * Copyright (C) 2017 NDLA
+ *
+ * See LICENSE
+ */
+
+package no.ndla.network
+
+import javax.servlet.http.HttpServletRequest
+
+import authentikat.jwt.JsonWebToken
+import com.typesafe.scalalogging.LazyLogging
+import no.ndla.network.model.AuthorizationException
+import org.json4s.native.Serialization.read
+
+import scala.util.{Failure, Success, Try}
+
+
+object User extends LazyLogging {
+  val AuthorizationHeader = "Authorization"
+
+  private val userId = new ThreadLocal[Option[String]]
+  private val userRoles = new ThreadLocal[List[String]]
+
+  def set(request: HttpServletRequest): Unit = {
+    extractClaims(request) match {
+      case Success(claims) =>
+        userId.set(extractUserId(claims))
+        userRoles.set(extractUserRoles(claims))
+
+      case Failure(err) =>
+        logger.info(err.getMessage, err)
+        userId.set(None)
+        userRoles.set(List.empty)
+    }
+  }
+
+  def get: Option[String] = userId.get
+  def getRoles: List[String] = userRoles.get
+
+  def clear(): Unit = {
+    userId.remove()
+    userRoles.remove()
+  }
+
+  private def extractClaims(request: HttpServletRequest): Try[JWTClaims] = {
+    implicit val formats = org.json4s.DefaultFormats
+
+    val optAuthHeader = Option(request.getHeader(AuthorizationHeader))
+    optAuthHeader.map(authHeader => {
+      val jwt = authHeader.replace("Bearer ", "")
+      jwt match {
+        case JsonWebToken(header, claimsSet, signature) => {
```

Comment:
Vi sjekker ikke signaturen her. Det er api-gateway sin oppgave, og det er den som har en secret som den bruker til å validere med. Her dekoder vi kun JWT-en for å hente ut informasjonen.